test/API/H5_api_dataset_test.c

```diff
@@ -4632,8 +5052,9 @@ test_read_multi_dataset_small_hyperslab(void)
     for (i = 0; i < DATASET_MULTI_COUNT; i++) {
         char dset_name[DSET_NAME_BUF_SIZE];
 
-        if (DSET_NAME_BUF_SIZE <= snprintf(dset_name, DSET_NAME_BUF_SIZE, "%s%zu%c",
-                                           DATASET_SMALL_READ_TEST_HYPERSLAB_DSET_NAME, i, '\0'))
+        if (DSET_NAME_BUF_SIZE <= (unsigned int)snprintf(dset_name, DSET_NAME_BUF_SIZE, "%s%zu%c",
```

Comment:
Same comment here about the unsigned cast (and probably other places in the code as well).